### DISCUSSION TITLE
Schema: ability to represent union types?

### DISCUSSION THREAD

**Discussion by DeaconDesperado** [kdl-org/kdl#371] [Q&A]
Consider the following KDL nodes:  semantically they are named colors, and they can be defined using:

* a hex string 
* a 1 digit 256 color code 
* a 3 digit Truecolor code

```
colorX "#79646A"
colorY 16
colorZ 255 184 108
```

Assuming the node name represents the color name, how could one validate for the union type of `string | number | (number, number, number)` using the schema definition?

**Comment by zkat** [comment]
There's no support for this right now, but now's a good time to spec it out. Do you want to give it a shot? I think you'd need to add a new `union` node that's basically identical to the `node`, but can specify different "shapes" of that node. I say `union`, because `enum` is already taken by the "these specific values" concept.

**Reply by DeaconDesperado** [reply]
Thanks!  I can make an attempt at it!
